### PR TITLE
2.4: fix build errors found with go 1.11 test

### PR DESCRIPTION
**Pull request by hmlanigan**
## Description of change

Fix build errors found with running go test with go version 1.11. 

## QA steps

Have branch downloaded and dependencies setup.
Install go 1.11
Edit Makefile to not run pre-check when check is run.
make check   <-- intermittent unit test failures are possible, but no build failures should be logged.

### CODE REVIEW

File: provider/maas/environ.go [hunk 1]
```diff
@@ -1723,7 +1723,7 @@ func (environ *maasEnviron) filteredSubnets(ctx context.ProviderCallContext, nod
 		}
 		subnetIdFloat, err := fields["id"].GetFloat64()
 		if err != nil {
-			return nil, errors.Annotatef(err, "cannot get subnet Id: %v")
+			return nil, errors.Annotatef(err, "cannot get subnet Id")
```
Comment (anastasiamac) [inline provider/maas/environ.go]: This can probably be just Annotate(

File: provider/ec2/environ.go [hunk 2]
```diff
@@ -1320,7 +1320,7 @@ func (e *environ) destroyControllerManagedEnvirons(ctx context.ProviderCallConte
 		// (anastasiamac 2018-03-21) This is strange - we do try
 		// to destroy all volumes but afterwards, if we have encountered any errors,
 		// we will return first one...The same logic happens on detach..?...
-		return errors.Annotatef(err, "destroying volume %q", volIds[i], err)
+		return errors.Annotatef(err, "destroying volume %q", volIds[i])
```
Comment (anastasiamac) [inline provider/ec2/environ.go]: Oh wow, yeah.. good call :)

File: provider/azure/instance.go [hunk 3]
```diff
@@ -283,7 +283,7 @@ func (inst *azureInstance) OpenPorts(ctx context.ProviderCallContext, machineId
 			nil, // abort channel
 		)
 		if err := <-errCh; err != nil {
-			return errors.Annotatef(err, "creating security rule for %s", securityRule)
+			return errors.Annotatef(err, "creating security rule for %+v", securityRule)
```
Comment (anastasiamac) [inline provider/azure/instance.go]: Why would we want %+v here? What was wrong with %s?

File: worker/caasoperatorprovisioner/worker.go [hunk 4]
```diff
@@ -114,7 +114,7 @@ func (p *provisioner) loop() error {
 			}
 			for app, password := range p.appPasswords {
 				if err := p.ensureOperator(app, password); err != nil {
-					return errors.Annotatef(err, "updating operator for q with new api addresses", app)
+					return errors.Annotatef(err, "updating operator for %q with new api addresses", app)
```
Comment (anastasiamac) [inline worker/caasoperatorprovisioner/worker.go]: Ha :) interesting.. thank you for the pick up!

File: provider/maas/environ.go (hunk 1 above)
Comment (hmlanigan) [reply provider/maas/environ.go]: updated

File: provider/azure/instance.go (hunk 3 above)
Comment (hmlanigan) [reply provider/azure/instance.go]: provider/azure/instance.go:286: Annotatef format %s has arg securityRule of wrong type github.com/Azure/azure-sdk-for-go/arm/network.SecurityRule

Looks like network.SecurityRule doesn't implement String()

Comment (hmlanigan) [reply provider/azure/instance.go]: i ended up changing this one to %s and ruleName, matched better with the other log and error messages.

**Comment by hmlanigan** [comment]
$$merge$$

**Comment by hmlanigan** [comment]
$$merge$$

**Pull request merged**